src/modules/cpu/kernel/fog.hpp

```diff
@@ -0,0 +1,1524 @@
+                        srcPtrChannel += srcDescPtr->strides.cStride;
+                        dstPtrChannel += dstDescPtr->strides.cStride;
+                    }
+                    srcPtrTemp++;
+                    dstPtrTemp++;
+                    fogAlphaMaskPtrTemp++;
+                    fogIntensityMaskPtrTemp++;
+                }
+
+                srcPtrRow += srcDescPtr->strides.hStride;
+                dstPtrRow += dstDescPtr->strides.hStride;
+                fogAlphaMaskPtrRow += fog_width;
+                fogIntensityMaskPtrRow += fog_width;
+            }
+        }
+    }
+
+    return RPP_SUCCESS;
+}
+
+RppStatus fog_f16_f16_host_tensor(Rpp16f *srcPtr,
+                                  RpptDescPtr srcDescPtr,
+                                  Rpp16f *dstPtr,
+                                  RpptDescPtr dstDescPtr,
+                                  RpptROIPtr roiTensorPtrSrc,
+                                  RpptRoiType roiType,
+                                  RppLayoutParams layoutParams,
+                                  rpp::Handle& handle)
+{
+    RpptROI roiDefault = {0, 0, (Rpp32s)srcDescPtr->w, (Rpp32s)srcDescPtr->h};
+    Rpp32u numThreads = handle.GetNumThreads();
+
+    omp_set_dynamic(0);
+#pragma omp parallel for num_threads(numThreads)
+    for(int batchCount = 0; batchCount < dstDescPtr->n; batchCount++)
+    {
+        RpptROI roi;
+        RpptROIPtr roiPtrInput = &roiTensorPtrSrc[batchCount];
+        compute_roi_validation_host(roiPtrInput, &roi, &roiDefault, roiType);
+
+
+        Rpp16f *srcPtrImage, *dstPtrImage;
+        srcPtrImage = srcPtr + batchCount * srcDescPtr->strides.nStride;
+        dstPtrImage = dstPtr + batchCount * dstDescPtr->strides.nStride;
+
+        Rpp32u bufferLength = roi.xywhROI.roiWidth * layoutParams.bufferMultiplier;
+
+        Rpp16f *srcPtrChannel, *dstPtrChannel;
+        srcPtrChannel = srcPtrImage + (roi.xywhROI.xy.y * srcDescPtr->strides.hStride) + (roi.xywhROI.xy.x * layoutParams.bufferMultiplier);
+        dstPtrChannel = dstPtrImage;
+
+        Rpp32u fog_width = 0;
+        Rpp32u fog_height = 0;
+        Rpp32f *fogAlphaMaskPtr, *fogIntensityMaskPtr;
+        if(roi.xywhROI.roiWidth<=224 && roi.xywhROI.roiHeight<=224)
+        {
+            fog_width = 224;
+            fog_height =224;
+            std::random_device rd;  // Random number engine seed
+            std::mt19937 gen(rd()); // Seeding rd() to fast mersenne twister engine
+            std::uniform_int_distribution<> distribX(0, fog_width - roi.xywhROI.roiWidth);
+            std::uniform_int_distribution<> distribY(0, fog_height - roi.xywhROI.roiHeight);
+
+            RppiPoint maskLoc;
+            maskLoc.x = distribX(gen);
+            maskLoc.y = distribY(gen);
+
+            fogAlphaMaskPtr = &fogAlphaMask_224_224[(fog_width * maskLoc.y) + maskLoc.x];
+            fogIntensityMaskPtr = &fogIntensityMask_224_224[(fog_width * maskLoc.y) + maskLoc.x];
```

Comment:
You can may be add a common helper function for setting this fogWidth, fogHeight, fogAlphaMaskPtr, fogIntensityMaskPtr values and use this helper functions for all U8, F16, F32. I8 codes

Reply:
I removed the code for mask choosing .